lib/vmdb/appliance.rb

```diff
@@ -36,6 +36,14 @@ def self.USER_AGENT
       "#{self.PRODUCT_NAME}/#{self.VERSION}".freeze
     end
 
+    def self.check_automate_disk_version_against_db(fh)
+      rh_domain = MiqAeDomain.find_by(:name => "RedHat")
```

Comment:
Should this be the "ManageIQ" domain?

Reply:
nope. "[RFE] How to tell if RedHat domain is out of sync after upgrade without UI"

Reply:
:disappointed: Hmm... I wonder if we can put this somewhere in the Red Hat plugin...

Reply:
That will be the fifth iteration of _where to even put this darn thing_ but I'm always up for suggestions.

Reply:
Does it make sense to have both checks, for both ManageIQ and Redhat domains?  Do we have the version and available_version information there?  If so, can we instead do something like this:

```ruby
# get top level domains
MiqAeDomain.where(:enabled => true, :parent_id => nil).each do |domain|
  if domain.version != domain.available_version
    fh.info("#{domain.name} on disk differs from domain in db:")
    fh.info("Current version - #{domain.version}, Available version - #{domain.available_version}")
  end
end
```

Reply:
Note, I have no idea how to check the top level domains, I made it up by looking at my db.  Enable + no parent id makes sense to me but who knows.

Reply:
@jrafanie It probably does make sense to include both, but I suspect that they will either both be in sync or out of sync.

Also, all `MiqAeDomain` instances are top-level objects as the domain is a special top-level namespace.

Reply:
Does it make sense to loop through all domains that are enabled? It would be nice to not have to hardcode a upstream/downstream string.

Reply:
We can loop over the `system` domains which should always be enabled so no need to check for that part.

`MiqAeDomain.where(:source => MiqAeDomain::SYSTEM_SOURCE)`